From e0190cc5945c5f56f4381e696b36adebc2ce1e30 Mon Sep 17 00:00:00 2001
From: Kyle Repinski <repinski23@gmail.com>
Date: Sun, 28 Feb 2016 15:28:55 -0600
Subject: [PATCH 06/13] hwc: Build with -Wall -Werror.

Also builds with Clang now too.
TODO: Go through and actually remove some of these unused parameters.

Change-Id: Ie69f88d9610e2c8eda1d795acbe44d259c9a2189
---
 hwc/Android.mk   |  2 +-
 hwc/dock_image.c |  1 -
 hwc/dock_image.h |  4 ++--
 hwc/hwc.c        | 24 +++++++++++++++++-------
 hwc/hwc_dev.h    |  4 ++--
 hwc/rgz_2d.c     |  9 +++------
 hwc/sw_vsync.c   |  1 -
 7 files changed, 25 insertions(+), 20 deletions(-)

diff --git a/hwc/Android.mk b/hwc/Android.mk
index dbaa6d4..d2d6b0b 100644
--- a/hwc/Android.mk
+++ b/hwc/Android.mk
@@ -22,7 +22,7 @@ LOCAL_STATIC_LIBRARIES := libpng
 LOCAL_MODULE_TAGS := optional
 
 LOCAL_MODULE := hwcomposer.omap4
-LOCAL_CFLAGS := -DLOG_TAG=\"ti_hwc\"
+LOCAL_CFLAGS := -DLOG_TAG=\"ti_hwc\" -Wall -Werror
 LOCAL_C_INCLUDES += external/libpng external/zlib
 
 LOCAL_C_INCLUDES += \
diff --git a/hwc/dock_image.c b/hwc/dock_image.c
index 6626fa7..f685ccd 100644
--- a/hwc/dock_image.c
+++ b/hwc/dock_image.c
@@ -47,7 +47,6 @@ static void free_png_image(image_info_t *img)
 
 static int load_png_image(char *path, image_info_t *img)
 {
-    void *ptr = NULL;
     static png_bytepp row_pointers = NULL; /* static prevents setjmp corruption. */
 
     FILE *fd = fopen(path, "rb");
diff --git a/hwc/dock_image.h b/hwc/dock_image.h
index 44a3271..ac53ea8 100644
--- a/hwc/dock_image.h
+++ b/hwc/dock_image.h
@@ -19,6 +19,8 @@
 
 #include <stdint.h>
 
+#include "display.h"
+
 /* ARGB image */
 struct image_info {
     int width;
@@ -29,8 +31,6 @@ struct image_info {
 };
 typedef struct image_info image_info_t;
 
-typedef struct omap_hwc_device omap_hwc_device_t;
-
 int init_dock_image(omap_hwc_device_t *hwc_dev, uint32_t max_width, uint32_t max_height);
 void load_dock_image();
 image_info_t *get_dock_image();
diff --git a/hwc/hwc.c b/hwc/hwc.c
index 2a7fff9..a4ff5a4 100644
--- a/hwc/hwc.c
+++ b/hwc/hwc.c
@@ -96,6 +96,9 @@ enum {
                     (f) == OMAP_DSS_COLOR_ARGB32 ? "ARGB32" : \
                     (f) == OMAP_DSS_COLOR_RGB16 ? "RGB565" : "??")
 
+//#define DUMP_LAYERS
+//#define DUMP_DSSCOMPS
+
 static bool debug = false;
 static bool debugpost2 = false;
 static bool debugblt = false;
@@ -129,6 +132,7 @@ static void showfps(void)
     }
 }
 
+#ifdef DUMP_LAYERS
 static void dump_layer(hwc_layer_1_t const* l)
 {
     ALOGD("\ttype=%d, flags=%08x, handle=%p, tr=%02x, blend=%04x, {%d,%d,%d,%d}, {%d,%d,%d,%d}",
@@ -142,7 +146,9 @@ static void dump_layer(hwc_layer_1_t const* l)
          l->displayFrame.right,
          l->displayFrame.bottom);
 }
+#endif
 
+#ifdef DUMP_DSSCOMPS
 static void dump_dsscomp(struct dsscomp_setup_dispc_data *d)
 {
     uint32_t i;
@@ -180,6 +186,7 @@ static void dump_dsscomp(struct dsscomp_setup_dispc_data *d)
                  (void *) oi->ba, (void *) oi->uv, c->stride);
     }
 }
+#endif
 
 struct dump_buf {
     char *buf;
@@ -467,12 +474,14 @@ static void setup_layer_base(struct dss2_ovl_cfg *oc, int index, uint32_t format
     oc->vc1.enable = 0;
 }
 
-static void setup_layer(omap_hwc_device_t *hwc_dev, struct dss2_ovl_info *ovl,
+static void setup_layer(omap_hwc_device_t *hwc_dev __unused, struct dss2_ovl_info *ovl,
                         hwc_layer_1_t *layer, int index, uint32_t format, int width, int height)
 {
     struct dss2_ovl_cfg *oc = &ovl->cfg;
 
-    //dump_layer(layer);
+#ifdef DUMP_LAYERS
+    dump_layer(layer);
+#endif
 
     setup_layer_base(oc, index, format, is_BLENDED(layer), width, height);
 
@@ -859,8 +868,8 @@ static int set_best_hdmi_mode(omap_hwc_device_t *hwc_dev, uint32_t xres, uint32_
 {
     int dis_ix = hwc_dev->on_tv ? 0 : 1;
     struct _qdis {
-        struct dsscomp_display_info dis;
         struct dsscomp_videomode modedb[MAX_DISPLAY_CONFIGS];
+        struct dsscomp_display_info dis; /* variable-sized type; should be at end of struct */
     } d = { .dis = { .ix = dis_ix } };
     omap_hwc_ext_t *ext = &hwc_dev->ext;
 
@@ -1158,7 +1167,6 @@ static inline int display_area(struct dss2_ovl_info *o)
 
 static int clone_layer(omap_hwc_device_t *hwc_dev, int ix) {
     struct dsscomp_setup_dispc_data *dsscomp = &hwc_dev->comp_data.dsscomp_data;
-    omap_hwc_ext_t *ext = &hwc_dev->ext;
     int ext_ovl_ix = dsscomp->num_ovls - hwc_dev->post2_layers;
     struct dss2_ovl_info *o = &dsscomp->ovls[dsscomp->num_ovls];
 
@@ -2016,7 +2024,9 @@ static int hwc_set(struct hwc_composer_device_1 *dev,
             }
         }
 
-        //dump_dsscomp(dsscomp);
+#ifdef DUMP_DSSCOMPS
+        dump_dsscomp(dsscomp);
+#endif
 
         // signal the event thread that a post has happened
         write(hwc_dev->pipe_fds[1], "s", 1);
@@ -2469,7 +2479,7 @@ static int hwc_query(struct hwc_composer_device_1* dev, int what, int* value)
 }
 
 static int hwc_eventControl(struct hwc_composer_device_1* dev,
-        int dpy, int event, int enabled)
+        int dpy __unused, int event, int enabled)
 {
     omap_hwc_device_t *hwc_dev = (omap_hwc_device_t *) dev;
 
@@ -2498,7 +2508,7 @@ static int hwc_eventControl(struct hwc_composer_device_1* dev,
     }
 }
 
-static int hwc_blank(struct hwc_composer_device_1 *dev, int dpy, int blank)
+static int hwc_blank(struct hwc_composer_device_1 *dev __unused, int dpy __unused, int blank __unused)
 {
     // We're using an older method of screen blanking based on
     // early_suspend in the kernel.  No need to do anything here.
diff --git a/hwc/hwc_dev.h b/hwc/hwc_dev.h
index 471a577..717e9e5 100644
--- a/hwc/hwc_dev.h
+++ b/hwc/hwc_dev.h
@@ -123,7 +123,6 @@ struct omap_hwc_device {
     pthread_mutex_t lock;
 
     IMG_framebuffer_device_public_t *fb_dev;
-    struct dsscomp_display_info fb_dis;
     int fb_fd;                   /* file descriptor for /dev/fb0 */
     int dsscomp_fd;              /* file descriptor for /dev/dsscomp */
     int hdmi_fb_fd;              /* file descriptor for /dev/fb1 */
@@ -174,7 +173,8 @@ struct omap_hwc_device {
     bool use_sw_vsync;
 
     display_t *displays[MAX_DISPLAYS];
+
+    struct dsscomp_display_info fb_dis; /* variable-sized type; should be at end of struct */
 };
-typedef struct omap_hwc_device omap_hwc_device_t;
 
 #endif
diff --git a/hwc/rgz_2d.c b/hwc/rgz_2d.c
index ac0c1a0..9614d50 100644
--- a/hwc/rgz_2d.c
+++ b/hwc/rgz_2d.c
@@ -495,7 +495,7 @@ static void rgz_rotate_src(struct rgz_blt_entry* e, int src_orientation, int is_
         swap(srcgeom->width, srcgeom->height);
 }
 
-static void rgz_set_src_data(rgz_out_params_t *params, rgz_layer_t *rgz_layer,
+static void rgz_set_src_data(rgz_out_params_t *params __unused, rgz_layer_t *rgz_layer,
     blit_rect_t *subregion_rect, struct rgz_blt_entry* e, int src_orientation,
     int is_src2)
 {
@@ -555,7 +555,7 @@ static void rgz_set_clip_rect(rgz_out_params_t *params, blit_rect_t *subregion_r
 /*
  * Configures blit entry to set src2 is the same as the destination
  */
-static void rgz_set_src2_is_dst(rgz_out_params_t *params, struct rgz_blt_entry* e)
+static void rgz_set_src2_is_dst(rgz_out_params_t *params __unused, struct rgz_blt_entry* e)
 {
     /* omaplfb is in charge of assigning the correct src2desc in the kernel */
     e->src2geom = e->dstgeom;
@@ -1099,7 +1099,6 @@ static void rgz_handle_dirty_region(rgz_t *rgz, rgz_in_params_t *params,
     for (i = 0; i < cur_fb_state->rgz_layerno; i++) {
         rgz_layer_t *cur_rgz_layer = &cur_fb_state->rgz_layers[i];
         rgz_layer_t *prev_rgz_layer = NULL;
-        int layer_changed = 0;
 
         if (i == 0) {
             /*
@@ -1500,9 +1499,7 @@ static int hal_to_ocd(int color)
     }
 }
 
-static BVFN_MAP bv_map;
 static BVFN_BLT bv_blt;
-static BVFN_UNMAP bv_unmap;
 
 static int rgz_handle_to_stride(IMG_native_handle_t *h)
 {
@@ -1773,7 +1770,7 @@ static struct rgz_blt_entry* rgz_blts_get(struct rgz_blts *blts, rgz_out_params_
     return ne;
 }
 
-static int rgz_blts_bvdirect(rgz_t *rgz, struct rgz_blts *blts, rgz_out_params_t *params)
+static int rgz_blts_bvdirect(rgz_t *rgz __unused, struct rgz_blts *blts, rgz_out_params_t *params __unused)
 {
     struct bvbatch *batch = NULL;
     int rv = -1;
diff --git a/hwc/sw_vsync.c b/hwc/sw_vsync.c
index fce6d37..ec95d9f 100644
--- a/hwc/sw_vsync.c
+++ b/hwc/sw_vsync.c
@@ -54,7 +54,6 @@ static void *vsync_loop(void *data)
     nsecs_t now = 0, period = vsync_rate, next_vsync = 0, next_fake_vsync = 0, sleep = 0;
     omap_hwc_device_t *hwc_dev = (omap_hwc_device_t *)data;
     tp_sleep.tv_sec = tp_sleep.tv_nsec = 0;
-    bool reset_timers = true;
 
     setpriority(PRIO_PROCESS, 0, HAL_PRIORITY_URGENT_DISPLAY);
 
-- 
2.7.4

